src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionTanInputViewController.swift

```diff
@@ -18,32 +18,57 @@
 import Foundation
 import UIKit
 
-class ExposureSubmissionTanInputViewController: UIViewController, SpinnerInjectable {
+class ExposureSubmissionTanInputViewController: UIViewController, ENANavigationControllerWithFooterChild, SpinnerInjectable {
 	// MARK: - Attributes.
 
 	@IBOutlet var scrollView: UIScrollView!
 	@IBOutlet var contentView: UIView!
 	@IBOutlet var descriptionLabel: UILabel!
 	@IBOutlet var errorLabel: UILabel!
 	@IBOutlet var errorView: UIView!
-	@IBOutlet var tanInput: ENATanInput!
+	@IBOutlet var tanInput: ENATanInput! { didSet { tanInput.delegate = self } }
 
 	var initialTan: String?
 	var exposureSubmissionService: ExposureSubmissionService?
 	var spinner: UIActivityIndicatorView?
 
 	// MARK: - View lifecycle methods.
 
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		navigationItem.title = AppStrings.ExposureSubmissionTanEntry.title
+
+		navigationFooterItem?.primaryButtonTitle = AppStrings.ExposureSubmissionTanEntry.submit
+		navigationFooterItem?.isPrimaryButtonEnabled = false
+
+		descriptionLabel.text = AppStrings.ExposureSubmissionTanEntry.description
+		errorView.alpha = 0
```

Comment:
What is the reason for setting the alpha like this...?

Reply:
This was part of a previous PR when the TAN input field was refactored. It enabled smooth animations for showing or hiding the error message.